lib/src/extensions/byte_stream.dart

```diff
@@ -60,19 +60,26 @@ extension ByteStreamExtensions on Stream<List<int>> {
   /// This treats [this] as a process that emits only stdout. If [this] emits an
   /// error, it's treated the same as an unhandled Dart error in a
   /// [Script.capture] block: it's printed to stderr and the virtual stream
-  /// process exits with error code 257.
+  /// process exits with error code `257`.
   ///
-  /// The script can only be interrupted by closing [this] stream. Any [signal]
-  /// is ignored and returns whether or not the script is sill running.
+  /// [Script.kill] returns `true` if the stream was interrupted and the script
+  /// exits with [Script.exitCode] `143`, or `false` if the stream was already
+  /// closed.
   Script get _asScript {
+    var signalCloser = StreamCloser<List<int>>();
     return Script.fromComponents("stream", () {
       var exitCodeCompleter = Completer<int>.sync();
       return ScriptComponents(
           NullStreamSink(),
-          onDone(() => exitCodeCompleter.complete(0)),
+          transform(signalCloser).onDone(() =>
+              exitCodeCompleter.complete(signalCloser.isClosed ? 143 : 0)),
           Stream.empty(),
-          exitCodeCompleter.future,
-          ([ProcessSignal? _]) => !exitCodeCompleter.isCompleted);
+          exitCodeCompleter.future, (_) async {
+        if (exitCodeCompleter.isCompleted) return false;
```

Comment:
We shouldn't ask callers to manually check for this; the `Script` simply shouldn't invoke `onSignal` after it's already emitted an exit code.

Reply:
That's right.